File: 007-ReverseInteger/C++/main.cpp
class Solution {
public:
    int reverse(int x) {               
        long long n = 0;
        while (x) {
            n = (x % 10) + n * 10;
            x /= 10;
        }
        
        return (n < INT_MIN || n > INT_MAX) ? 0 : n;
    }
};

File: 007-ReverseInteger/Go/main.go
func reverse(x int) int {
    if x < 0 {
        return -reverse(-x)
    }
    
    var n int
    for x != 0 {
        n = (x % 10) + n * 10
        x /= 10
    }
    
    if n > (1 << 31) {
        return 0
    }
    
    return n
}